packages/contracts/src/token/CBL.sol

```diff
@@ -0,0 +1,30 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.19;
+
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { ERC20Permit } from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+import { ERC20Burnable } from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import { Pausable } from "@openzeppelin/contracts/utils/Pausable.sol";
+
+contract CBL is ERC20, Ownable, ERC20Permit, ERC20Burnable, Pausable {
+    error CBL__MaxSupplyExceeded();
+
+    uint256 public maxSupply;
+
+    constructor(address _owner, uint256 _maxSupply) ERC20("Credbull", "CBL") ERC20Permit("Credbull") Ownable(_owner) {
+        maxSupply = _maxSupply;
+    }
+
+    function mint(address to, uint256 amount) external onlyOwner {
+        if (totalSupply() + amount > maxSupply) {
+            revert CBL__MaxSupplyExceeded();
+        }
+        _mint(to, amount);
+    }
+
+    function burn(address from, uint256 amount) external onlyOwner {
+        _burn(from, amount);
+    }
```

Comment:
suggest we keep to the burn approach from OpenZeppelin.  this is not a restricted operation.  anyone can choose to burn their own tokens.  if we wish to restrict, we need to override update.
```
    /**
     * @dev Destroys a `value` amount of tokens from `account`, lowering the total supply.
     * Relies on the `_update` mechanism.
     *
     * Emits a {Transfer} event with `to` set to the zero address.
     *
     * NOTE: This function is not virtual, {_update} should be overridden instead
     */
    function _burn(address account, uint256 value) internal {
        if (account == address(0)) {
            revert ERC20InvalidSender(address(0));
        }
        _update(account, address(0), value);
    }
```